### DISCUSSION TITLE
Noobaa with private s3 storage

### DISCUSSION THREAD

**Discussion by dkoci** [noobaa/noobaa-core#6827] [Q&A] [answered]
Hi.
I dont have any experience with noobaa. First of all, is noobaa for freee?
Is it possible to use noobaa with private s3 storage (instead of public, e.g. AWS)? I would like to deploy projectquay on our openshift cluster. We have private s3 storage which would be perfect for quay, but I am a bit confused about noobaa.

**Comment by nimrod-becker** [comment, marked as answer]
NooBaa is open-sourced and can be used for free.
Yes, you can connect NooBaa to any S3 compatible service (this is also how its being labeled in the UI and CRDs).

You might want to check the operator repo Readme for a quick start with NooBaa.
https://github.com/noobaa/noobaa-operator

**Reply by dkoci** [reply]
Thank you for your answer. I have already deployed noobaa (using noobaa-operator). It is running but I am a bit confused how to use it.

**Reply by guymguym** [reply]
Hi @dkoci 
What brought you to try noobaa? Are you looking for any feature in particular? Are you planning to use multiple clusters/clouds?
Would be great if you can share info on your applications and operations that you use with S3.

In general noobaa provides several interfaces:
1. CLI (noobaa-operator) - look around in `noobaa --help`
2. CRD (noobaa-operator)
3. RPC-API (noobaa-core internal)
4. S3 (noobaa-core)

Noobaa is a managed S3 service, where admins can apply a configuration per bucket (per bucket-class) and connect to various backends such as S3, Azure Blob, GCS, and Filesystems. I'll share below a simple diagram that might help to visualize this. Also this article might give more color - https://next.redhat.com/2021/07/27/bucket-caching-for-kubernetes/

![image](https://user-images.githubusercontent.com/1091422/144494303-cae882ac-7634-4c45-b3c7-a82827e7e161.png)

**Reply by dkoci** [reply]
Thank you for reply. As I mentioned I noticed word noobaa in projectquay documentation. I would like to deploy porjectquay to Openshift. Projectquay requires object storage which we have in our company. I have experience with kubernetes/openshift, but not with noobaa, so there are alot of things i need to learn :)
